sync ddl failed in cdc/binlog downstream

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
upstream
```
prepare stmt from "create table ttt(a int primary key)";
execute stmt;
```
<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)
cdc/binlog downstream
```
mysql> show tables;
+--------------+
| Tables_in_cp |
+--------------+
| ttt          |
+--------------+
1 row in set (0.04 sec)
```
### 3. What did you see instead (Required)
```
mysql> show tables;
Empty set (0.05 sec)
```

binlog log
```
[2020/05/26 11:39:52.174 +08:00] [ERROR] [load.go:732] ["exec failed"] [sql="execute stmt"] [error="Error 8111: Prepared statement not found"] [errorVerbose="Error 8111: Prepared statement not found\ngithub.com/pingcap/errors.AddStack\n\t/home/zyguan/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20190809092503-95897b64e011/errors.go:174\ngithub.com/pingcap/errors.Trace\n\t/home/zyguan/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20190809092503-95897b64e011/juju_adaptor.go:15\ngithub.com/pingcap/tidb-binlog/pkg/loader.(*loaderImpl).execDDL\n\t/home/zyguan/tidb-binlog/pkg/loader/load.go:420\ngithub.com/pingcap/tidb-binlog/pkg/loader.(*batchManager).execDDL\n\t/home/zyguan/tidb-binlog/pkg/loader/load.go:730\ngithub.com/pingcap/tidb-binlog/pkg/loader.(*batchManager).put\n\t/home/zyguan/tidb-binlog/pkg/loader/load.go:753\ngithub.com/pingcap/tidb-binlog/pkg/loader.(*loaderImpl).Run\n\t/home/zyguan/tidb-binlog/pkg/loader/load.go:605\ngithub.com/pingcap/tidb-binlog/drainer/sync.(*MysqlSyncer).run\n\t/home/zyguan/tidb-binlog/drainer/sync/mysql.go:232\nruntime.goexit\n\t/usr/local/lib/go1.13.5/src/runtime/asm_amd64.s:1357"]
[2020/05/26 11:39:52.174 +08:00] [INFO] [load.go:853] ["txnManager has been closed"]
[2020/05/26 11:39:52.174 +08:00] [INFO] [load.go:547] ["{16 20 0xc000a33a40 0xc000a461c0 false 1 true true}"]
[2020/05/26 11:39:52.174 +08:00] [INFO] [load.go:548] ["Run()... in Loader quit"]
[2020/05/26 11:39:52.174 +08:00] [INFO] [mysql.go:228] ["Successes chan quit"]
[2020/05/26 11:39:52.174 +08:00] [INFO] [syncer.go:249] ["handleSuccess quit"]
[2020/05/26 11:39:52.174 +08:00] [INFO] [load.go:797] ["run()... in txnManager quit"]
[2020/05/26 11:39:52.174 +08:00] [ERROR] [syncer.go:457] ["Failed to close syncer"] [error="Error 8111: Prepared statement not found"] [errorVerbose="Error 8111: Prepared statement not found\ngithub.com/pingcap/errors.AddStack\n\t/home/zyguan/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20190809092503-95897b64e011/errors.go:174\ngithub.com/pingcap/errors.Trace\n\t/home/zyguan/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20190809092503-95897b64e011/juju_adaptor.go:15\ngithub.com/pingcap/tidb-binlog/pkg/loader.(*loaderImpl).execDDL\n\t/home/zyguan/tidb-binlog/pkg/loader/load.go:420\ngithub.com/pingcap/tidb-binlog/pkg/loader.(*batchManager).execDDL\n\t/home/zyguan/tidb-binlog/pkg/loader/load.go:730\ngithub.com/pingcap/tidb-binlog/pkg/loader.(*batchManager).put\n\t/home/zyguan/tidb-binlog/pkg/loader/load.go:753\ngithub.com/pingcap/tidb-binlog/pkg/loader.(*loaderImpl).Run\n\t/home/zyguan/tidb-binlog/pkg/loader/load.go:605\ngithub.com/pingcap/tidb-binlog/drainer/sync.(*MysqlSyncer).run\n\t/home/zyguan/tidb-binlog/drainer/sync/mysql.go:232\nruntime.goexit\n\t/usr/local/lib/go1.13.5/src/runtime/asm_amd64.s:1357"]
[2020/05/26 11:39:52.174 +08:00] [ERROR] [server.go:289] ["syncer exited abnormal"] [error="Error 8111: Prepared statement not found"] [errorVerbose="Error 8111: Prepared statement not found\ngithub.com/pingcap/errors.AddStack\n\t/home/zyguan/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20190809092503-95897b64e011/errors.go:174\ngithub.com/pingcap/errors.Trace\n\t/home/zyguan/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20190809092503-95897b64e011/juju_adaptor.go:15\ngithub.com/pingcap/tidb-binlog/pkg/loader.(*loaderImpl).execDDL\n\t/home/zyguan/tidb-binlog/pkg/loader/load.go:420\ngithub.com/pingcap/tidb-binlog/pkg/loader.(*batchManager).execDDL\n\t/home/zyguan/tidb-binlog/pkg/loader/load.go:730\ngithub.com/pingcap/tidb-binlog/pkg/loader.(*batchManager).put\n\t/home/zyguan/tidb-binlog/pkg/loader/load.go:753\ngithub.com/pingcap/tidb-binlog/pkg/loader.(*loaderImpl).Run\n\t/home/zyguan/tidb-binlog/pkg/loader/load.go:605\ngithub.com/pingcap/tidb-binlog/drainer/sync.(*MysqlSyncer).run\n\t/home/zyguan/tidb-binlog/drainer/sync/mysql.go:232\nruntime.goexit\n\t/usr/local/lib/go1.13.5/src/runtime/asm_amd64.s:1357"]
[2020/05/26 11:39:52.174 +08:00] [INFO] [util.go:72] [Exit] [name=syncer]
[2020/05/26 11:39:52.174 +08:00] [INFO] [server.go:451] ["begin to close drainer server"]
[2020/05/26 11:39:52.178 +08:00] [INFO] [server.go:416] ["has already update status"] [id=172.16.4.213:8249]
[2020/05/26 11:39:52.178 +08:00] [INFO] [server.go:455] ["commit status done"]
[2020/05/26 11:39:52.178 +08:00] [INFO] [collector.go:135] ["publishBinlogs quit"]
[2020/05/26 11:39:52.178 +08:00] [INFO] [pump.go:77] ["pump is closing"] [id=172.16.4.210:8250]
[2020/05/26 11:39:52.178 +08:00] [INFO] [pump.go:77] ["pump is closing"] [id=172.16.4.203:8250]
[2020/05/26 11:39:52.178 +08:00] [INFO] [util.go:72] [Exit] [name=heartbeat]
[2020/05/26 11:39:52.178 +08:00] [INFO] [util.go:72] [Exit] [name=collect]
[2020/05/26 11:39:52.179 +08:00] [INFO] [main.go:73] ["drainer exit"]
```

cdc log
```
[2020/05/26 10:56:05.396 +08:00] [WARN] [mysql.go:137] ["execute DDL with error, retry later"] [query="execute stmt"] [error="Error 8111: Prepared statement not found"] [errorVerbose="Error 8111: Prepared statement not found\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20190809092503-95897b64e011/errors.go:174\ngithub.com/pingcap/errors.Trace\n\tgithub.com/pingcap/errors@v0.11.5-0.20190809092503-95897b64e011/juju_adaptor.go:15\ngithub.com/pingcap/ticdc/cdc/sink.(*mysqlSink).execDDL\n\tgithub.com/pingcap/ticdc@/cdc/sink/mysql.go:165\ngithub.com/pingcap/ticdc/cdc/sink.(*mysqlSink).execDDLWithMaxRetries.func1\n\tgithub.com/pingcap/ticdc@/cdc/sink/mysql.go:128\ngithub.com/pingcap/ticdc/pkg/retry.Run.func1\n\tgithub.com/pingcap/ticdc@/pkg/retry/retry.go:31\ngithub.com/cenkalti/backoff.RetryNotify\n\tgithub.com/cenkalti/backoff@v2.2.1+incompatible/retry.go:37\ngithub.com/cenkalti/backoff.Retry\n\tgithub.com/cenkalti/backoff@v2.2.1+incompatible/retry.go:24\ngithub.com/pingcap/ticdc/pkg/retry.Run\n\tgithub.com/pingcap/ticdc@/pkg/retry/retry.go:30\ngithub.com/pingcap/ticdc/cdc/sink.(*mysqlSink).execDDLWithMaxRetries\n\tgithub.com/pingcap/ticdc@/cdc/sink/mysql.go:126\ngithub.com/pingcap/ticdc/cdc/sink.(*mysqlSink).EmitDDLEvent\n\tgithub.com/pingcap/ticdc@/cdc/sink/mysql.go:121\ngithub.com/pingcap/ticdc/cdc.(*changeFeed).handleDDL\n\tgithub.com/pingcap/ticdc@/cdc/changefeed.go:446\ngithub.com/pingcap/ticdc/cdc.(*Owner).handleDDL\n\tgithub.com/pingcap/ticdc@/cdc/owner.go:359\ngithub.com/pingcap/ticdc/cdc.(*Owner).run\n\tgithub.com/pingcap/ticdc@/cdc/owner.go:599\ngithub.com/pingcap/ticdc/cdc.(*Owner).Run\n\tgithub.com/pingcap/ticdc@/cdc/owner.go:536\ngithub.com/pingcap/ticdc/cdc.(*Server).campaignOwnerLoop\n\tgithub.com/pingcap/ticdc@/cdc/server.go:167\ngithub.com/pingcap/ticdc/cdc.(*Server).run.func1\n\tgithub.com/pingcap/ticdc@/cdc/server.go:199\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20190911185100-cd5d95a43a6e/errgroup/errgroup.go:57\nruntime.goexit\n\truntime/asm_amd64.s:1357"]
[2020/05/26 10:56:05.671 +08:00] [WARN] [mysql.go:137] ["execute DDL with error, retry later"] [query="execute stmt"] [error="Error 8111: Prepared statement not found"] [errorVerbose="Error 8111: Prepared statement not found\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20190809092503-95897b64e011/errors.go:174\ngithub.com/pingcap/errors.Trace\n\tgithub.com/pingcap/errors@v0.11.5-0.20190809092503-95897b64e011/juju_adaptor.go:15\ngithub.com/pingcap/ticdc/cdc/sink.(*mysqlSink).execDDL\n\tgithub.com/pingcap/ticdc@/cdc/sink/mysql.go:165\ngithub.com/pingcap/ticdc/cdc/sink.(*mysqlSink).execDDLWithMaxRetries.func1\n\tgithub.com/pingcap/ticdc@/cdc/sink/mysql.go:128\ngithub.com/pingcap/ticdc/pkg/retry.Run.func1\n\tgithub.com/pingcap/ticdc@/pkg/retry/retry.go:31\ngithub.com/cenkalti/backoff.RetryNotify\n\tgithub.com/cenkalti/backoff@v2.2.1+incompatible/retry.go:37\ngithub.com/cenkalti/backoff.Retry\n\tgithub.com/cenkalti/backoff@v2.2.1+incompatible/retry.go:24\ngithub.com/pingcap/ticdc/pkg/retry.Run\n\tgithub.com/pingcap/ticdc@/pkg/retry/retry.go:30\ngithub.com/pingcap/ticdc/cdc/sink.(*mysqlSink).execDDLWithMaxRetries\n\tgithub.com/pingcap/ticdc@/cdc/sink/mysql.go:126\ngithub.com/pingcap/ticdc/cdc/sink.(*mysqlSink).EmitDDLEvent\n\tgithub.com/pingcap/ticdc@/cdc/sink/mysql.go:121\ngithub.com/pingcap/ticdc/cdc.(*changeFeed).handleDDL\n\tgithub.com/pingcap/ticdc@/cdc/changefeed.go:446\ngithub.com/pingcap/ticdc/cdc.(*Owner).handleDDL\n\tgithub.com/pingcap/ticdc@/cdc/owner.go:359\ngithub.com/pingcap/ticdc/cdc.(*Owner).run\n\tgithub.com/pingcap/ticdc@/cdc/owner.go:599\ngithub.com/pingcap/ticdc/cdc.(*Owner).Run\n\tgithub.com/pingcap/ticdc@/cdc/owner.go:536\ngithub.com/pingcap/ticdc/cdc.(*Server).campaignOwnerLoop\n\tgithub.com/pingcap/ticdc@/cdc/server.go:167\ngithub.com/pingcap/ticdc/cdc.(*Server).run.func1\n\tgithub.com/pingcap/ticdc@/cdc/server.go:199\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20190911185100-cd5d95a43a6e/errgroup/errgroup.go:57\nruntime.goexit\n\truntime/asm_amd64.s:1357"]
```

### 4. Affected version (Required)

<!-- v3.0.0, v4.0.0, etc -->
```
Release Version: v4.0.0-rc.2-12-g6098373
Edition: Community
Git Commit Hash: 6098373c11a94eeb114a6c106f297e3b1498b158
Git Branch: HEAD
UTC Build Time: 2020-05-19 04:19:09
GoVersion: go1.13.7
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false
```

### 5. Root Cause Analysis

<!-- should be filled by the investigator before it's closed -->

--- a/executor/adapter.go
+++ b/executor/adapter.go
@@ -718,6 +718,7 @@ func (a *ExecStmt) buildExecutor() (Executor, error) {
 		if err != nil {
 			return nil, err
 		}
+		a.Ctx.SetValue(sessionctx.QueryString, executorExec.stmt.Text())
 		a.OutputNames = executorExec.outputNames
 		a.isPreparedStmt = true
 		a.Plan = executorExec.plan

--- a/session/session_test.go
+++ b/session/session_test.go
@@ -314,6 +314,18 @@ func (s *testSessionSuite) TestQueryString(c *C) {
 	c.Assert(err, IsNil)
 	qs := tk.Se.Value(sessionctx.QueryString)
 	c.Assert(qs.(string), Equals, "CREATE TABLE t2(id bigint PRIMARY KEY, age int)")
+
+	// Test execution of DDL through the "Execute" interface.
+	_, err = tk.Se.Execute(context.Background(), "use test;")
+	c.Assert(err, IsNil)
+	_, err = tk.Se.Execute(context.Background(), "drop table t2")
+	c.Assert(err, IsNil)
+	_, err = tk.Se.Execute(context.Background(), "prepare stmt from 'CREATE TABLE t2(id bigint PRIMARY KEY, age int)'")
+	c.Assert(err, IsNil)
+	_, err = tk.Se.Execute(context.Background(), "execute stmt")
+	c.Assert(err, IsNil)
+	qs = tk.Se.Value(sessionctx.QueryString)
+	c.Assert(qs.(string), Equals, "CREATE TABLE t2(id bigint PRIMARY KEY, age int)")
 }
 
 func (s *testSessionSuite) TestAffectedRows(c *C) {

